query/src/sql/planner/binder/select.rs

```diff
@@ -139,20 +147,36 @@ impl<'a> Binder {
     }
 
     #[async_recursion]
-    pub(crate) async fn bind_query(
+    pub(crate) async fn bind_set_expr(
         &mut self,
         bind_context: &BindContext,
-        query: &Query,
+        set_expr: &SetExpr,
+        order_by: &[OrderByExpr],
     ) -> Result<(SExpr, BindContext)> {
-        let (mut s_expr, bind_context) = match &query.body {
-            SetExpr::Select(stmt) => {
-                self.bind_select_stmt(bind_context, stmt, &query.order_by)
-                    .await
-            }
+        match set_expr {
+            SetExpr::Select(stmt) => self.bind_select_stmt(bind_context, stmt, order_by).await,
```

Comment:
The semantic of `ORDER BY` will become much different if we support set operations.

We should add some tests about this, for example:

```sql
create table t(a int);

select a as b from t order by a; -- allowed
select a as b from t union select b from t order by a; -- not allowed
```

Reply:
added